postgresql_extensible crash on null value

### Relevant telegraf.conf:
```
[[outputs.influxdb]]
  urls = ["http://localhost:8086"]

  [[inputs.postgresql_extensible.query]]
    sqlquery='''
      SELECT *
      FROM pg_catalog.pg_stat_activity
    '''
    version=100
    withdbname=false
```

### System info:

```
Linux 4.4.0-1061-aws #70-Ubuntu SMP Fri May 25 21:47:34 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

telegraf --version
Telegraf v1.7.4 (git: release-1.7 578db7ef)
```

### Steps to reproduce:

drop target measurement on influxdb side and

select have to capture at least one column with null value, can be simplified:
```
    sqlquery='''
      SELECT
        null as datname
      FROM pg_catalog.pg_stat_activity
'''
```

### Expected behavior:
Insert null and did not crash
### Actual behavior:

```
# /usr/bin/telegraf -config /etc/telegraf/fr_usd_dev-5432.conf
panic: interface conversion: interface {} is nil, not string

goroutine 16 [running]:
github.com/influxdata/telegraf/plugins/inputs/postgresql_extensible.(*Postgresql).accRow(0xc42019e120, 0xc420304cc1, 0x10, 0x17ac460, 0xc4201c8780, 0x17c9f20, 0xc42048dce0, 0xc4203b8c00, 0x2, 0x2, ...)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/postgresql_extensible/postgresql_extensible.go:226 +0xe5e
github.com/influxdata/telegraf/plugins/inputs/postgresql_extensible.(*Postgresql).Gather(0xc42019e120, 0x17c9f20, 0xc42048dce0, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/postgresql_extensible/postgresql_extensible.go:184 +0x40e
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc42017c540, 0xc42040cf00, 0x17c9f20, 0xc42048dce0)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:153 +0x47
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
        /go/src/github.com/influxdata/telegraf/agent/agent.go:152 +0xd4

```
### Additional info:
crash is not written to logfile.

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible.go
@@ -19,14 +19,8 @@ type Postgresql struct {
 	postgresql.Service
 	Databases      []string
 	AdditionalTags []string
-	Query          []struct {
-		Sqlquery    string
-		Version     int
-		Withdbname  bool
-		Tagvalue    string
-		Measurement string
-	}
-	Debug bool
+	Query          query
+	Debug          bool
 }
 
 type query []struct {
@@ -221,7 +215,7 @@ func (p *Postgresql) accRow(meas_name string, row scanner, acc telegraf.Accumula
 		return err
 	}
 
-	if *columnMap["datname"] != nil {
+	if _, ok := columnMap["datname"]; ok {
 		// extract the database name from the column map
 		dbname.WriteString((*columnMap["datname"]).(string))
 	} else {

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible.go
@@ -215,9 +215,9 @@ func (p *Postgresql) accRow(meas_name string, row scanner, acc telegraf.Accumula
 
 	if c, ok := columnMap["datname"]; ok && *c != nil {
 		// extract the database name from the column map
-		switch (*c).(type) {
+		switch datname := (*c).(type) {
 		case string:
-			dbname.WriteString((*columnMap["datname"]).(string))
+			dbname.WriteString(datname)
 		default:
 			dbname.WriteString("postgres")
 		}

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible.go
@@ -121,7 +121,6 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 	)
 
 	// Retreiving the database version
-
 	query = `select substring(setting from 1 for 3) as version from pg_settings where name='server_version_num'`
 	if err = p.DB.QueryRow(query).Scan(&db_version); err != nil {
 		db_version = 0
@@ -129,7 +128,6 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 
 	// We loop in order to process each query
 	// Query is not run if Database version does not match the query version.
-
 	for i := range p.Query {
 		sql_query = p.Query[i].Sqlquery
 		tag_value = p.Query[i].Tagvalue
@@ -215,9 +213,14 @@ func (p *Postgresql) accRow(meas_name string, row scanner, acc telegraf.Accumula
 		return err
 	}
 
-	if _, ok := columnMap["datname"]; ok {
+	if c, ok := columnMap["datname"]; ok && *c != nil {
 		// extract the database name from the column map
-		dbname.WriteString((*columnMap["datname"]).(string))
+		switch (*c).(type) {
+		case string:
+			dbname.WriteString((*columnMap["datname"]).(string))
+		default:
+			dbname.WriteString("postgres")
+		}
 	} else {
 		dbname.WriteString("postgres")
 	}

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible.go
@@ -221,7 +221,7 @@ func (p *Postgresql) accRow(meas_name string, row scanner, acc telegraf.Accumula
 		return err
 	}
 
-	if columnMap["datname"] != nil {
+	if *columnMap["datname"] != nil {
 		// extract the database name from the column map
 		dbname.WriteString((*columnMap["datname"]).(string))
 	} else {

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible_test.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible_test.go
@@ -223,3 +223,21 @@ func TestPostgresqlIgnoresUnwantedColumns(t *testing.T) {
 		assert.False(t, acc.HasMeasurement(col))
 	}
 }
+
+func TestAccRow(t *testing.T) {
+	p := Postgresql{}
+	var acc testutil.Accumulator
+	row := fakeRow{}
+	columns := []string{"this", "that", "thing"}
+
+	err := p.accRow("pgTEST", row, &acc, columns)
+	if err != nil {
+		fmt.Println("THISNADA", err)
+	}
+}
+
+type fakeRow struct{}
+
+func (t fakeRow) Scan(dest ...interface{}) error {
+	return nil
+}

--- a/plugins/inputs/postgresql_extensible/postgresql_extensible_test.go
+++ b/plugins/inputs/postgresql_extensible/postgresql_extensible_test.go
@@ -1,6 +1,7 @@
 package postgresql_extensible
 
 import (
+	"errors"
 	"fmt"
 	"testing"
 
@@ -227,17 +228,37 @@ func TestPostgresqlIgnoresUnwantedColumns(t *testing.T) {
 func TestAccRow(t *testing.T) {
 	p := Postgresql{}
 	var acc testutil.Accumulator
-	row := fakeRow{}
-	columns := []string{"this", "that", "thing"}
+	columns := []string{"datname", "cat"}
 
-	err := p.accRow("pgTEST", row, &acc, columns)
-	if err != nil {
-		fmt.Println("THISNADA", err)
+	testRows := []fakeRow{
+		{fields: []interface{}{1, "gato"}},
+		{fields: []interface{}{nil, "gato"}},
+		{fields: []interface{}{"name", "gato"}},
+	}
+	for i := range testRows {
+		err := p.accRow("pgTEST", testRows[i], &acc, columns)
+		if err != nil {
+			t.Fatalf("Scan failed: %s", err)
+		}
 	}
 }
 
-type fakeRow struct{}
+type fakeRow struct {
+	fields []interface{}
+}
 
-func (t fakeRow) Scan(dest ...interface{}) error {
+func (f fakeRow) Scan(dest ...interface{}) error {
+	if len(f.fields) != len(dest) {
+		return errors.New("Nada matchy buddy")
+	}
+
+	for i, d := range dest {
+		switch d.(type) {
+		case (*interface{}):
+			*d.(*interface{}) = f.fields[i]
+		default:
+			return fmt.Errorf("Bad type %T", d)
+		}
+	}
 	return nil
 }

